### PR TITLE
Adds another date format to parse dates from CDB.

### LINKED ISSUE #917: Breakpoint.getFinalTime() coming with two different formats from the server
e.g.,
2016-09-21T15:39:21Z
2016-09-21T15:57:04.000Z

a ParseException is being logged when the first one is returned.

The change was already acknowledged by the apis team, and the milliseconds bit is going away. In the meantime, we're interchangeably getting dates in two different formats.

### PR DESCRIPTION
**Pull request by joaoandremartins**
Fixes #917 
@etanshaul @patflynn @aslo

### CODE REVIEW

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java [hunk 1]
```diff
@@ -84,11 +86,17 @@ public static Date parseDateTime(@Nullable String dateString) {
     dateString = dateString.replaceAll("Z$", "-0000");
 
     SimpleDateFormat iso8601Format = new SimpleDateFormat(ISO_8601_FORMAT);
+    iso8601Format.setLenient(true);
 
     try {
       return iso8601Format.parse(dateString);
```
Comment (patflynn) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java]: If NO_MS is the new format that's being rolled out, should we check that first?

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java [hunk 2]
```diff
@@ -84,11 +86,17 @@ public static Date parseDateTime(@Nullable String dateString) {
     dateString = dateString.replaceAll("Z$", "-0000");
 
     SimpleDateFormat iso8601Format = new SimpleDateFormat(ISO_8601_FORMAT);
+    iso8601Format.setLenient(true);
```
Comment (patflynn) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java]: apparently this is the default.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java [hunk 3]
```diff
@@ -84,11 +86,17 @@ public static Date parseDateTime(@Nullable String dateString) {
...
 
     try {
```
Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java]: Might be overkill but what do you think about refactoring to something like this:

```
public static final String supportedFormats = Arrays.asList(ISO_.., ISO..);
..
for (String formatString : formats) {
  try {
   ...
   return format.parse(dateString);
  } catch (..) {
   LOG.error("Failed to parse datetime %s using format %s", ..);
  }
}
```

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java (hunk 2 above)
Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java]: This shouldn't have been here, forgot to delete it.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java (hunk 3 above)
Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java]: I would prefer your way if we had more than 2 formats. Realistically, we'll have two for a short time and then back to one. Which reminds me, I should also add a TODO to check on the no_ms rollout status and eventually clean up.

Comment (etanshaul) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java]: Agree I guess. I just don't really like the nested try catch but it's not a big deal.

**Comment by aslo** [comment]
:+1:

**Comment by etanshaul** [comment]
Aside form @patflynn's comments :ship: (although I prefer getting rid of the nested try catch's but I'll leave it up to you :)  )

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java (hunk 1 above)
Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java]: Makes sense.

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java [hunk 4]
```diff
@@ -85,16 +91,14 @@ public static Date parseDateTime(@Nullable String dateString) {
...
+    for (String format : FORMATS) {
       try {
-        iso8601Format = new SimpleDateFormat(ISO_8601_FORMAT_NO_MS);
-        return iso8601Format.parse(dateString);
-      } catch (ParseException pe2) {
-        LOG.error("error parsing datetime " + dateString, pe2);
+        dateFormat = new SimpleDateFormat(format);
+        return dateFormat.parse(dateString);
+      } catch (ParseException pe) {
+        LOG.warn("error parsing datetime " + dateString + " with format " + format);
```
Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/BreakpointUtil.java]: Added log.error.

**Comment by patflynn** [comment]
why did you ahve to push a commit to force a jenkins build?  Did it have a transient failure?

**Comment by joaoandremartins** [comment]
Yeah Jenkins is coming up with weird errors that don't seem related to this PR.
First, a CloudSdkPanelTest error was coming up, now it's a different one..

**Pull request merged**